pkg/lockfile/parse-yarn-lock.go

```diff
@@ -208,6 +254,33 @@ func tryExtractCommit(resolution string) string {
 	return ""
 }
 
+func buildDependencyTree(rootPkgName, rootPkgTargetVersion, rootPkgRegistry string, dependencies map[string]YarnPackage, packagesIndex map[string]*PackageDetails) []*PackageDetails {
```

Comment:
The cognitive complexity of this function is slightly high, maybe add a comment explaining what this does? Doesn't have to be too long